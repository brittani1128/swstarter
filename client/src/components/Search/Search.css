.search-form {
  width: 410px;
  height: 230px;
  border-radius: 4px;
  box-shadow: 0 1px 2px 0 rgba(132, 132, 132, 0.75);
  background-color: #ffffff;
  padding: 30px;
}

h6 {
  width: 205px;
  height: 18px;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 600;
  color: #383838;
  margin-top: 0;
}
.flex li {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}

.radio-btns {
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 30px;
}

.radio-btns li {
  padding-right: 30px;
}

ul {
  padding: 0px;
}

#input {
  width: 350px;
  height: 40px;
  box-shadow: inset 0 1px 3px 0 rgba(132, 132, 132, 0.75);
  background-color: #ffffff;
}
#search {
  width: 350px;
  border-radius: 20px;
  background-color: #0ab463;
  color: white;
  margin-top: 20px;
  height: 34px;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: bold;
  color: #ffffff;
  text-transform: uppercase;
}
