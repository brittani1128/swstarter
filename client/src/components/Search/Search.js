import React from "react";
import "./Search.css";
import { AppContext } from "../../utils/AppContext";

// Search form
class Search extends React.Component {
  static contextType = AppContext;

  render() {
    const context = this.context;
    return (
      <div className="search-form">
        <h2>What are you searching for?</h2>
        <form onSubmit={context.handleSubmit}>
          <ul className="flex">
            <div className="radio-btns">
              <li>
                <input
                  type="radio"
                  id="people"
                  name="searchQuery"
                  value="people"
                  onChange={context.handleSelect}
                />
                <label htmlFor="people">People</label>
              </li>
              <li>
                <input
                  type="radio"
                  id="movies"
                  name="searchQuery"
                  value="movies"
                  onChange={context.handleSelect}
                />
                <label htmlFor="movies">Movies</label>
              </li>
              <li>
                <input
                  type="radio"
                  id="planets"
                  name="searchQuery"
                  value="planets"
                  onChange={context.handleSelect}
                />
                <label htmlFor="planets">Planets</label>
              </li>
            </div>

            <li>
              <input
                type="text"
                name="input"
                id="input"
                value={context.resultSearch}
                onChange={context.handleInputChange}
                placeholder="e.g. Chewbacca, Yoda, Boba Fett"
              />
            </li>
            <li>
              <input
                type="button"
                name="search"
                id="search"
                value="Search"
                onClick={context.handleSubmit}
              />
            </li>
          </ul>
        </form>
      </div>
    );
  }
}

export default Search;
