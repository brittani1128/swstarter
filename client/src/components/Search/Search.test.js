import React from "react";
import { shallow } from "enzyme";
import Search from "./Search";

describe("Search component", () => {
  it("should render", () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });

  //   it("should render form without throwing an error", () => {
  //     expect(
  //       shallow(<Search />)
  //         .find(".search-form")
  //         .exists()
  //     ).toBe(true);
  //   });

  it("should render radio buttons", () => {
    expect(shallow(<Search />).find("input.radio").length).toEqual(4);
  });

  it("should render search input box", () => {
    expect(shallow(<Search />).find("#input").length).toEqual(1);
  });
});
