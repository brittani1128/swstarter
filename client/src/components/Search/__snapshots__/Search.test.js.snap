// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render 1`] = `
<div
  className="search-form"
>
  <h2>
    What are you searching for?
  </h2>
  <form>
    <ul
      className="flex"
    >
      <div
        className="radio-btns"
      >
        <li>
          <input
            className="radio"
            id="people"
            name="searchQuery"
            type="radio"
            value="people"
          />
          <label
            htmlFor="people"
          >
            People
          </label>
        </li>
        <li>
          <input
            className="radio"
            id="movies"
            name="searchQuery"
            type="radio"
            value="movies"
          />
          <label
            htmlFor="movies"
          >
            Movies
          </label>
        </li>
        <li>
          <input
            className="radio"
            id="planets"
            name="searchQuery"
            type="radio"
            value="planets"
          />
          <label
            htmlFor="planets"
          >
            Planets
          </label>
        </li>
        <li>
          <input
            className="radio"
            id="species"
            name="searchQuery"
            type="radio"
            value="species"
          />
          <label
            htmlFor="species"
          >
            Species
          </label>
        </li>
      </div>
      <li>
        <input
          id="input"
          name="input"
          placeholder="e.g. Chewbacca, Yoda, Boba Fett"
          type="text"
        />
      </li>
      <li>
        <input
          id="search"
          name="search"
          type="button"
          value="Search"
        />
      </li>
    </ul>
  </form>
</div>
`;
